### DISCUSSION TITLE
Where are the instructions for adding dataset to LAAC-LSCP on GIN?

### DISCUSSION THREAD

**Discussion by Valentina-Fed** [LAAC-LSCP/ChildProject#347] [Q&A] [answered]
@alecristia 
https://childproject.readthedocs.io/en/latest/vandam.html#publish-to-gin this explains how to publish to GIN, but where are the information about procedures? 
Thank you!

**Comment by lucasgautheron** [comment, marked as answer]
Hello!

You can find procedures for adding new datasets here: https://github.com/LAAC-LSCP/datalad-procedures

In your case I recommend using the EL1000 procedure with the following settings:

```bash
export GIN_ORGANIZATION='LAAC-LSCP'
export CONFIDENTIAL_DATASET=0
datalad create -c el1000 <dataset-name>
```

You will need owner permissions for that on LAAC-LSCP's gin organization

**Reply by alecristia** [reply]
sorry for the delay! I've a few questions:

- I understand the difference between LAAC1 and LAAC2, but not between LAAC2 and EL1000 -- perhaps the README can include at the top of the EL1000 description what this procedure sets up as default, like the other two (LAAC1/2) do?
- Right now, members of LAAC are working on 3 distinct cases: 

1) a dataset with Naomi (so she should be owner); 
2) tsimane2018 (so LAAC can be owner, but Camila should too); and 
3) Tsay (which is a public dataset, but the creator is still alive, though I'm not sure she'd like to take on the responsibility of being owner AND the dataset is already available in CHILDES). 

- For the first case, perhaps do we need instructions for a new procedure, or adapt EL1000? For the second, EL1000 or LAAC2? For the final one, should it be pushed to a public OSF repo rather than a private GIN one?

Tagging relevant people here: @Valentina-Fed @camiscaff @avishavish1999

**Reply by alecristia** [reply]
sorry, one more: how bad is it if I choose the wrong setup procedure (eg EL1000 rather than LAAC2)?

**Reply by alecristia** [reply]
I'm getting output that seems to suggest a bug:
`datalad create -c el1000 tsimanem2018`
>[INFO   ] Creating a new annex repo at /Users/alejandrinacristia/gitrepos/tsimanem2018 
[INFO   ] scanning for unlocked files (this may take some time) 
[WARNING] Cannot find procedure with name 'cfg_el1000' [run_procedure(cfg_el1000)] 

Also, the instructions say that I should create 2 repos on GIN: "Using the browser capabilities on GIN, create two empty repositories in your GIN organization: <dataset-name> and <dataset-name>-confidential, e.g. dataset1 and dataset1-confidential. Here's an example of creation of the first (i.e. non confidential); notice that (a) you need to create the repo from the organization (and not your personal account) and (b) you need to uncheck the box at the bottom during actual creation."

This is wrong or inaccurate (and I suspect in both cases, my fault): Either we only need to create one (and the procedure creates the second one) OR we only need to create two *when there is confidential content*

**Reply by lucasgautheron** [reply]
- Difference between EL1000 and LAAC2:
   - EL1000 configures 1 or 2 siblings (2 if there should be a confidential one) on the same organization
   - LAAC2 configures 3 siblings: el1000, el1000-confidential, and private (the latter being hosted on LAAC-LSCP and having all the data)
 
 - the EL1000 procedure should work even with users. I think all she has to do is specify her username instead of "EL1000" as the organization.
 - The correct interpretation re the instructions is "we only need to create two when there is confidential content"
 - Regarding the issue you mention (cannot find procedure EL1000); If you did run the install script before, I suspect you are having the same issue as Camila: https://github.com/LAAC-LSCP/datalad-procedures/issues/4 . TL;DR: this happens when there are several installations of DataLad and the script installs the procedures for a different version than the one that is invoked from the command line. In order to fix this, do `datalad run-procedure --discover`, copy the /directory/ of the yoda procedure in your clipboard, then do `python install.py --destination <the directory>`

I'll try to find a workaround for this issue.
I'll clarify the docs and the different options.
Also, because of a new feature from GIN, I might be able to automize the creation of the GIN repositories too: https://docs.datalad.org/en/latest/generated/man/datalad-create-sibling-gin.html

**Reply by alecristia** [reply]
Thanks for that! turns out - I hadn't installed procedures at all 🤕 

This feels like having to pick one's path through a forest - but perhaps this is just because there are few of us users out there, and eventually this will all clarify into well-worn paths...

**Comment by lucasgautheron** [comment]
(Though thinking about it, it would be interesting to know what's in this dataset -- e.g. does it include the audio, or just the annotations?)

**Reply by Valentina-Fed** [reply]
Thank you! It includes also audio.

**Comment by alecristia** [comment]
it's a public dataset, Tsay from childes.talkbank.org

**Comment by lucasgautheron** [comment]
Alright! Well, that should work!
